linera-views/src/journaling.rs

```diff
@@ -230,72 +234,73 @@ where
     /// Writes blocks to the database and resolves them later.
     async fn write_journal(
         &self,
-        simplified_batch: K::SimpBatch,
+        simplified_batch: K::Batch,
         base_key: &[u8],
     ) -> Result<JournalHeader, K::Error> {
         let mut iter = simplified_batch.into_iter();
-        let mut value_size = 0;
-        let mut transact_size = 0;
-        let mut simp_batch = K::SimpBatch::default();
+        let mut block = K::Batch::default();
+        let mut block_size = 0;
         let mut block_count = 0;
-        let mut transacts = K::SimpBatch::default();
+        let mut transaction = K::Batch::default();
+        let mut transaction_size = 0;
         loop {
-            let result = simp_batch.try_append(&mut iter, &mut value_size)?;
-            if !result {
+            if !iter.write_next_value(&mut block, &mut block_size)? {
                 break;
             }
```

Comment:
```suggestion
        while iter.write_next_value(&mut block, &mut block_size)? {
```

Reply:
So it's still a bit messy. We're not going to break here unless `iter` was empty from the beginning.